tests/integration/targets/ndo_template/tasks/main.yml

```diff
@@ -0,0 +1,592 @@
+          - query_new_tenant_policy_template_id.current.displayName == 'ansible_tenant_policy_changed'
+          - query_all_tenant is not changed
+          - query_all_tenant.current | length >= 2
+          - query_all is not changed
+          - query_all.current | length >= 8
+
+    # ERRORS
+
+    - name: Error on unknown template
+      cisco.mso.ndo_template:
+        <<: *update_tenant_policy_template
+        template_id: "unknown_id"
+        state: present
+      register: err_unknown_template_id
+      ignore_errors: true
+    
+    - name: Error on not allowed input site
+      cisco.mso.ndo_template:
+        <<: *update_tenant_policy_template
+        sites:
+          - name: '{{ mso_site | default("ansible_test") }}'
+        state: present
+      register: err_not_allowed_site_id
+      ignore_errors: true
+
+    - name: Error on not allowed template_type change
+      cisco.mso.ndo_template:
+        <<: *create_l3out_policy_template
+        sites: '{{ fakevar | default(omit) }}'
+        template_id: '{{ nm_create_new_l3out_policy_template.current.templateId }}'
+        template_type: tenant
+        state: present
+      register: err_not_allowed_template_type_change
+      ignore_errors: true
+
+    - name: Error on not allowed site change
+      cisco.mso.ndo_template:
+        <<: *create_l3out_policy_template
+        sites: 
+          - name: ansible_test_2
+        state: present
+      register: err_not_allowed_site_change
+      ignore_errors: true
+
+    - name: Error on not allowed site change
+      cisco.mso.ndo_template:
+        <<: *create_l3out_policy_template
+        sites:
+          - name: ansible_test 
+          - name: ansible_test_2
+        state: present
+      register: err_not_allowed_multiple_sites
+      ignore_errors: true
+
+    - name: Error on not allowed tenant change
+      cisco.mso.ndo_template:
+        <<: *create_l3out_policy_template
+        tenant: ansible_test_2
+        state: present
+      register: err_not_allowed_tenant_change
+      ignore_errors: true
+    
+    - name: Error on tenant must be provided
+      cisco.mso.ndo_template:
+        <<: *create_l3out_policy_template
+        name: l3out_policy_template_error
+        tenant: '{{ fakevar | default(omit)}}'
+        state: present
+      register: err_tenant_not_provided
+      ignore_errors: true
+    
+    - name: Error on site must be provided
+      cisco.mso.ndo_template:
+        <<: *create_l3out_policy_template
+        name: l3out_policy_template_error
+        sites: '{{ fakevar | default(omit)}}'
+        state: present
+      register: err_site_not_provided
+      ignore_errors: true
+
+    - name: Error on not allowed tenant
+      cisco.mso.ndo_template:
+        <<: *create_monitoring_access_template
+        tenant: ansible_test
+        state: present
+      register: err_not_allowed_tenant
+      ignore_errors: true
+
+    - name: Err on template query without template_type
```

Comment:
Err -> Error

Reply:
Changed and rebased all the PRs that are affected by this